src/ledger/fuzz.zig

```diff
@@ -0,0 +1,466 @@
+            defer dataMap.clearAndFree();
+            for (startKey..endKey) |key| {
+                // Fill the buffer with random bytes for each key.
+                for (0..buffer.len) |i| {
+                    buffer[i] = @intCast(random.int(u8));
+                }
+
+                const value: []const u8 = try allocator.dupe(u8, buffer[0..]);
+                const data = Data{ .value = value };
+
+                try batch.put(cf1, key, data);
+                try dataMap.put(@as(u32, @intCast(key)), data);
+            }
+            // Commit batch put.
+            try db.commit(&batch);
+            var it = dataMap.iterator();
+            while (it.next()) |entry| {
+                const entryKey = entry.key_ptr.*;
+                const expected = entry.value_ptr.*;
+                const actual = try db.get(
+                    allocator,
+                    cf1,
+                    entryKey,
+                ) orelse return error.KeyNotFoundError;
+                try std.testing.expect(std.mem.eql(u8, expected.value, actual.value));
+            }
+        }
+
+        // Batch delete.
+        {
+            const startKey = random.int(u32);
+            const endKey = startKey +| @as(u32, random.int(u8));
+            var buffer: [61]u8 = undefined;
+            var batch = try db.initWriteBatch();
+            defer batch.deinit();
+            for (startKey..endKey) |key| {
+                // Fill the buffer with random bytes for each key.
+                for (0..buffer.len) |i| {
+                    buffer[i] = @intCast(random.int(u8));
+                }
+
+                const value: []const u8 = try allocator.dupe(u8, buffer[0..]);
+                const data = Data{ .value = value };
+
+                try batch.put(cf1, key, data);
+                try batch.delete(cf1, key);
+            }
+            // Commit batch put and delete.
+            try db.commit(&batch);
+            for (startKey..endKey) |key| {
+                const actual = try db.get(allocator, cf1, @as(u32, @intCast(key)));
+                try std.testing.expectEqual(null, actual);
+            }
+        }
+
+        // Batch delete range.
+        {
+            const startKey = random.int(u32);
+            const endKey = startKey +| @as(u32, random.int(u8));
+            var buffer: [61]u8 = undefined;
+            var batch = try db.initWriteBatch();
+            defer batch.deinit();
+            for (startKey..endKey) |key| {
+                // Fill the buffer with random bytes for each key.
+                for (0..buffer.len) |i| {
+                    buffer[i] = @intCast(random.int(u8));
+                }
+
+                const value: []const u8 = try allocator.dupe(u8, buffer[0..]);
+                const data = Data{ .value = value };
+
+                try batch.put(cf1, key, data);
+            }
+            try batch.deleteRange(cf1, startKey, endKey);
+            // Commit batch put and delete range.
```

Comment:
if these methods dont modify the dataMap - then we should probably add a note saying there is no confirmation in these methods

Reply:
[0688ef9](https://github.com/Syndica/sig/pull/452/commits/0688ef9b24d019d4b10ab23d0ccdebcdf0d9a7dd)